external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java

```diff
@@ -0,0 +1,66 @@
+package com.stnetix.ariaddna.externalcloudapi.cloudinterface;
+
+/**
+ * This is an abstract interface containing general methods for manipulation with any external cloud storage.
+ * Every certain cloud storage realization must implement this interface.
+ */
+public interface iAbstractCloud {
+    /**
+     * An uploadBinFile method implementation takes as a parameter a reference to a file-chunk.
+     * After that it should initiate a transferring procedure to an external cloud storage.
+     * @return The method implementation should return the amount of successfully sent bytes orex
+     * zero if the transfer fails.
+     */
+
+    //TODO look for possibility of a cloud provider to return successfully sent byte amount.
+    //TODO Function should take as a parameter a ref to a file chunk: Type?
+    int uploadBinFile();
+
+    /**
+     * A downloadBinFile method implementation is responsible for downloading a file-chunk from
+     * a cloud storage to user's local storage.
+     */
+    //TODO what should be a parameter and a return value?
+    void downloadBinFile();
+
+    /**
+     * A createDirectory implementation method allows one to create a new folder or directory
+     * at a cloud storage.
+     * @return A method implementation should signal whether a directory was successfully created (true) or not (false).
+     */
+    //FIXME param?
+    boolean createDirectory();
```

Comment:
For example yandex disk api method to create directory has String parameter with path of creating folder. Response was link of creating folder.
My opinion - we pass String with path and get String with URL. This response URL we can use to generate VUSF snapshot.